src/braket/experimental/autoqasm/program/program.py

```diff
@@ -341,15 +341,26 @@ def _free_symbol_names(expr: FreeParameterExpression) -> Iterable[str]:
         """
         return sorted([str(s) for s in expr._expression.free_symbols if isinstance(s, Symbol)])
 
-    def register_parameter(self, parameter_name: str) -> None:
+    def register_parameter(
+        self, parameter_name: str, parameter_type: Optional[Union[float, int]] = float
+    ) -> None:
         """Register an input parameter if it has not already been registered.
-        Only floats are currently supported.
 
         Args:
             parameter_name (str): The name of the parameter to register with the program.
+            parameter_type (Optional[Union[float, int]]): The type of the parameter to register
+                with the program.
         """
         if parameter_name not in self._free_parameters:
-            self._free_parameters[parameter_name] = oqpy.FloatVar("input", name=parameter_name)
+            if parameter_type in [float, None]:
```

Comment:
This isn't being called by users, do we need to support `parameter_type=None`? The default is `float` not `None`

Reply:
Actually, if we check for type collision with existing free parameters, there is a difference between no type specified and float, so maybe None would be a more suitable default if we want to allow code like
```
@aq.main
def func(a: int):
    for i in aq.range(FreeParameter("a")):
        ...
```
default float would raise an exception